### PR TITLE
feat: added MARC-Record Page

### PR DESCRIPTION
**Pull request by gd-salle**
- Added AddMARCRecord Page
- Added MARCSection Component
- Fixed Routing of Pages and created a route folder for better file structure
- Fixed Modal Alignment

### CODE REVIEW

File: src/components/Verify/VerifyUser.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  const navigate = useNavigate();
+
+  const handleSubmit = async (): Promise<void> => {
+    setError(null);
+    setLoading(true);
+
+    try {
+      // Simulated API call
+      const data = await new Promise((resolve) =>
+        setTimeout(() => resolve({ id: 1, emailAdd: 'example@unc.edu.ph' }), 1000)
```
Comment (Benz0407) [inline src/components/Verify/VerifyUser.tsx:21]: This is causing the sending of OTP not working. Please, check Catalog branch recent push.

Comment (Benz0407) [reply src/components/Verify/VerifyUser.tsx:21]: Correct code snippet:
const handleSubmit = async (): Promise<void> => {
    // e.preventDefault();
    setError(null);
    setLoading(true);

    try {
      const data = await UserService.verifyUser(studentId);
      // Check if the data contains student information, meaning verification was successful
      if (data && data.id) {
        // If the OTP is sent, navigate to the OTP verification page with the studentId
        localStorage.setItem('emailAdd', data.emailAdd);
        navigate('/verify-otp');
      } else {
        // If the student is not found or enrolled, show an error
        setError('Student not found or not currently enrolled.');
      }
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
    } catch (error) {
      setError('Error verifying student ID.');
    } finally {
      // Set loading to false when the process completes
      setLoading(false);
    }
  };

File: src/routes/UserRoutes.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Route, Routes } from 'react-router-dom';
+import LibraryCardGenerationPage from '../pages/LibraryCardGeneration/LibraryCardGenerationPage';
+import ProtectedRoute from './ProtectedRoute';
+
+const UserRoutes = () => (
+  <Routes>
+    <Route
+      element={<ProtectedRoute allowedRoles={['LIBRARIAN']} redirectPath="/" />}
```
Comment (Benz0407) [inline src/routes/UserRoutes.tsx:8]: ALL USER ROLES SHOULD BE ABLE TO ACCESS THIS

File: src/components/Modal/LoginModal/Modal.tsx [hunk 3]
```diff
@@ -131,19 +127,23 @@ const Login: React.FC<LoginProps> = ({ open, onClose }) => {
...
-              Sign up
-            </Button>
-          </Typography>
+  Don't have an account?{' '}
+  <Button
+    color="inherit"
+    sx={{ color: '#d32f2f', textDecoration: 'underline' }}
+    onClick={() => {
+      onClose(); // Close the login modal
+      navigate('/register');
```
Comment (Benz0407) [inline src/components/Modal/LoginModal/Modal.tsx:136]: This should be redirected to the Verify User before can access the registration form

**Review by Benz0407 — APPROVED** [review]

File: src/components/Modal/LoginModal/Modal.tsx (hunk 3 above)
Comment (gd-salle) [reply src/components/Modal/LoginModal/Modal.tsx:136]: since this is only a 1 line change, will do after the merge.

**Pull request merged**